src/manifest_parser.cc

```diff
@@ -148,14 +148,17 @@ bool ManifestParser::ParseRule(string* err) {
   while (lexer_.PeekToken(Lexer::INDENT)) {
     string key;
     EvalString value;
-    if (!ParseLet(&key, &value, err))
+    if (!ParseLet(&key, &value, err)) {
+      delete rule;
```

Comment:
It seems you forgot lines 168 and 170 below :-/

I suggest using an `std::unique_ptr<Rule>` local variable definition on line 146 above, then passing its ownership on line 175 with `env->AddRule(rule.release())` to prevent similar issues to crop up in the future.

Reply:
Great idea! This also reflect the to-do comment from 2010 to use a `scoped_ptr`.